src/pfe/portal/modules/Templates.js

```diff
@@ -287,11 +276,28 @@ function getTemplateStyle(template) {
   return template.projectStyle || 'Codewind';
 }
 
-// reduce function, want to take repository list and index
-// them by url, and use that to avoid processing duplicate entries
-function reducer(repos, repo) {
-  if (repo.url && !repos[repo.url])
-    repos[repo.url] = repo;
+async function getReposFromProviders(providers) {
+  const repos = [];
+  await Promise.all(providers.map(async(provider) => {
+    try {
+      const providedRepos = await provider.getRepositories();
+      providedRepos.forEach(repo => {
```

Comment:
I guess you're using the `catch` here for the case where a bad provider returns something that's not an array?

Reply:
I actually meant the `catch` to catch errors in `provider.getRepositories()`, but yea it implicitly catches cases when the provider returns a non-array

Reply:
we could do `if (!Array.isArray(providedRepos)) throw new Error(<friendly error message>);`

Reply:
updated:
<img width="1270" alt="image" src="https://user-images.githubusercontent.com/18170169/62950536-22761a00-bde0-11e9-8371-9d0c70b6bf02.png">